File: problems/leetcode/26-remove_duplicates_from_sorted_array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length == 0) return 0;

        int lastUniqueIndex = 0;

        for (int i = 1; i < nums.length; i++){
            if (nums[i] != nums[lastUniqueIndex]){
                lastUniqueIndex++;
                nums[lastUniqueIndex] = nums[i];
            }
        }
        return lastUniqueIndex + 1;
    }
}

File: leetcode/26-remove_duplicates_from_sorted_array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        
        unique_count = 1
        
        for i in range(1, len(nums)):
            if nums[i] != nums[i - 1]:
                nums[unique_count] = nums[i]
                unique_count += 1
        
        return unique_count